### PR TITLE
III-6357-Ownerships page

### PR DESCRIPTION
**Pull request by vhande**
### Added
- Ownerships page
- Ownerships table
- Icon times circle
- Icon arrow left

### Fixed

- Alert full width

---

Ticket: https://jira.uitdatabank.be/browse/III-6357

### CODE REVIEW

File: src/pages/organizers/[organizerId]/ownerships/index.page.tsx [hunk 1]
```diff
@@ -1,22 +1,160 @@
...
+              </Stack>
+            )}
+          </Stack>
+
+          <Stack spacing={3} flex={1}>
+            <Button
+              variant={ButtonVariants.PRIMARY}
+              css={`
+                padding: 10px 15px !important;
+              `}
```
Comment (Anahkiasen) [inline src/pages/organizers/[organizerId]/ownerships/index.page.tsx:117]: Would maybe just use the `padding` prop here unless there was an issue

File: src/pages/organizers/[organizerId]/ownerships/index.page.tsx [hunk 2]
```diff
@@ -1,22 +1,160 @@
...
+
+  const organizerId = useMemo(
+    () => router.query.organizerId as string,
+    [router.query.organizerId],
+  );
+  const getOrganizerByIdQuery = useGetOrganizerByIdQuery({
+    id: organizerId,
+  });
+  // @ts-expect-error
+  const organizer = getOrganizerByIdQuery?.data;
```
Comment (Anahkiasen) [inline src/pages/organizers/[organizerId]/ownerships/index.page.tsx:36]: I had forgotten how much it bothered me that our queries lost the types of the things :D Can we maybe still do `organizer: Organizer =` for extra safety

**Review by Anahkiasen — CHANGES_REQUESTED** [review]

File: src/pages/organizers/[organizerId]/ownerships/OwnershipsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  renderActions: (request: OwnershipRequest) => JSX.Element;
+};
+
+export const OwnershipsTable = ({ requests, renderActions }: Props) => {
+  const getGlobalValue = getValueFromTheme('global');
+  const { grey3 } = colors;
+  const { t } = useTranslation();
+  return (
+    <Stack
+      width="100%"
```
Comment (simon-debruijn) [inline src/pages/organizers/[organizerId]/ownerships/OwnershipsTable.tsx:21]: Doesn't flex 1 give 100% width already

File: src/pages/organizers/[organizerId]/ownerships/OwnershipsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      flex={1}
+      padding={4}
+      css={`
+        box-shadow: ${getGlobalValue('boxShadow.medium')};
+        border-radius: 0.5rem;
+        background-color: white;
+        margin-bottom: 2rem;
```
Comment (simon-debruijn) [inline src/pages/organizers/[organizerId]/ownerships/OwnershipsTable.tsx:28]: would be better added with props

File: src/pages/organizers/[organizerId]/ownerships/OwnershipsTable.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      `}
+    >
+      <Inline
+        justifyContent="space-between"
+        css={`
+          border-bottom: 1px solid ${grey3};
+          padding-bottom: 0.5rem;
```
Comment (simon-debruijn) [inline src/pages/organizers/[organizerId]/ownerships/OwnershipsTable.tsx:35]: would be better added with props

File: src/pages/organizers/[organizerId]/ownerships/index.page.tsx [hunk 6]
```diff
@@ -1,22 +1,148 @@
...
+  });
+
+  const requestsByState: { [key: string]: OwnershipRequest[] } = useMemo(
+    () =>
+      groupBy(
+        // @ts-expect-error
+        getOwnershipRequestsQuery.data?.member,
+        'state',
+      ),
+    [getOwnershipRequestsQuery],
```
Comment (simon-debruijn) [inline src/pages/organizers/[organizerId]/ownerships/index.page.tsx:51]:
```suggestion
    [getOwnershipRequestsQuery.data],
```

File: src/pages/organizers/[organizerId]/ownerships/index.page.tsx [hunk 7]
```diff
@@ -1,22 +1,148 @@
...
+                  )}
+                />
+              </Stack>
+            )}
+          </Stack>
+          <Stack spacing={3.5} flex={1}>
+            <Button variant={ButtonVariants.PRIMARY}>
+              {t('organizers.ownerships.actions.add')}
+            </Button>
+            <Button
```
Comment (simon-debruijn) [inline src/pages/organizers/[organizerId]/ownerships/index.page.tsx:112]: You probably want to use a Link with button primary styling here

File: src/pages/organizers/[organizerId]/ownerships/index.page.tsx [hunk 8]
```diff
@@ -1,22 +1,148 @@
...
+    await useGetOrganizerByIdQuery({
+      req,
+      queryClient,
+      id: query.organizerId,
+    });
+    await useGetOwnershipRequestsQuery({
+      req,
+      queryClient,
+      organizerId: query.organizerId,
+    });
```
Comment (simon-debruijn) [inline src/pages/organizers/[organizerId]/ownerships/index.page.tsx:138]:
```suggestion
    await Promise.all(
    [
    await useGetOrganizerByIdQuery({
      req,
      queryClient,
      id: query.organizerId,
    }),
    await useGetOwnershipRequestsQuery({
      req,
      queryClient,
      organizerId: query.organizerId,
    })
    ]
    )
```

File: src/ui/Alert.tsx [hunk 9]
```diff
@@ -130,60 +130,57 @@ const Alert = ({
   ...props
 }: AlertProps) => {
   return (
-    <Inline
+    <Stack
       role="alert"
       alignSelf={fullWidth ? 'normal' : 'flex-start'}
       display={visible ? 'flex' : 'none'}
       {...getInlineProps(props)}
```
Comment (simon-debruijn) [inline src/ui/Alert.tsx:137]:
```suggestion
      {...getStackProps(props)}
```

**Review by simon-debruijn — CHANGES_REQUESTED** [review]

File: src/pages/organizers/[organizerId]/ownerships/index.page.tsx (hunk 7 above)
Comment (vhande) [reply src/pages/organizers/[organizerId]/ownerships/index.page.tsx:112]: This is not within the scope of this ticket.

**Review by Anahkiasen — APPROVED** [review]

**Pull request merged**